base/compiler/ssair/inlining.jl

```diff
@@ -1100,11 +1095,8 @@ function assemble_inline_todo!(ir::IRCode, sv::OptimizationState)
         if signature_fully_covered && length(cases) == 0 && only_method isa Method
             if length(splits) > 1
                 # get match information for a single overall match instead of union splits
-                meth = get(sv.matching_methods_cache, sig.atype) do
-                    ms = _methods_by_ftype(sig.atype, sv.params.MAX_METHODS,
-                        sv.world, UInt[typemin(UInt)], UInt[typemin(UInt)])
-                    return ms
-                end
+                (meth, min_valid, max_valid) =
```

Comment:
Seems like we should be doing something with these values (`update_valid_age!(min_valid, max_valid, sv)`) if analyze_method is successful